File: GeneratedEntityReaderClasses/Classes/ceng_CColorFloat.cs
namespace GeneratedNoitaClasses;
public class ceng_CColorFloat :  iNoitaType<ceng_CColorFloat>{
	public NoitaFloat r { get; set; } = new();
	public NoitaFloat g { get; set; } = new();
	public NoitaFloat b { get; set; } = new();
	public NoitaFloat a { get; set; } = new();

    public void Read(NoitaStream s){
		r.Read(s);
		g.Read(s);
		b.Read(s);
		a.Read(s);

    }
    public void Write(NoitaStream s){
		r.Write(s);
		g.Write(s);
		b.Write(s);
		a.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/ceng_CColorFloat.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaFloat

from NoitaStream import NoitaStream
@dataclass
class ceng_CColorFloat:
    r:NoitaFloat = field(default_factory=NoitaFloat)
    g:NoitaFloat = field(default_factory=NoitaFloat)
    b:NoitaFloat = field(default_factory=NoitaFloat)
    a:NoitaFloat = field(default_factory=NoitaFloat)
    
    def Read(self, s:NoitaStream):
        self.r.Read(s)
        self.g.Read(s)
        self.b.Read(s)
        self.a.Read(s)
    
    def Write(self, s:NoitaStream):
        self.r.Write(s)
        self.g.Write(s)
        self.b.Write(s)
        self.a.Write(s)

